tools/visinity/vis-postprocess.xml

```diff
@@ -0,0 +1,69 @@
+<tool id="visinity_postprocess" name="Visualize Visinity Archive" tool_type="interactive" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="22.01">
+    <description>Visualize pre-processed Visinity archives</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+    <expand macro="stdio"/>
+    <expand macro="version_cmd"/>
+    <entry_points>
+        <entry_point name="Visualize Visinity Archive" requires_domain="True">
+            <port>8000</port>
+            <url>/</url>
+        </entry_point>
+    </entry_points>
+    <command detect_errors="aggressive">
+        <![CDATA[
+    cp working/inputs.json /visinity/ &&
+    cp "$__tool_directory__/archive_ingest.py" /visinity/archive_ingest.py &&
+    cd /visinity/ &&
+    CONDAFY="micromamba run /bin/bash -c" &&
+    \$CONDAFY "python archive_ingest.py inputs.json" &&
+    \$CONDAFY "python run.py";
+        ]]>
+    </command>
+    <configfiles>
+	    <inputs name='inputs' filename='inputs.json' data_style='paths'/>
+    </configfiles>
+    <inputs>
+        <param label='Visinity Archive' name='vis_archive' type='data' format='zip' help="Output of Create Visinity Archive tool"/>
+        <param label='Registered Images' name='channel_files' type='data_collection' collection_type='list' format='ome.tiff' help="Collection of properly formatted OME-TIFF images (Example: output of ASHLAR)"/>
+        <param label='Segmentation Masks' name='label_files' type='data_collection' collection_type='list' format='tiff' help="Collection of masks (tiff) as generated by cell/nuclear segmentation tool (Mesmer, S3segmenter, etc.)"/>
+    </inputs>
+    <outputs>
+        <data name="output" format="txt"/>
+    </outputs>
+    <tests>
+    </tests>
+    <help><![CDATA[
+**Visualize a Visinity Archive**
+
+Running Visinity in Galaxy requires execution of two separate tools in succession:
+
+1. Create Visinity Archive 
```

Comment:
Just out of interest how long does the first tool take? Is it computational intensive?

Reply:
there are a couple reasons for the two step process:

1. as you suspected, time/compute is a factor; as an example, a medium sized invocation on an 8vCPU/64GB RAM took about 15 minutes. with the two step process, you only have to create the archive once, then the IT can start visualizations immediately, and it can be re-visualized at a later time without repeating this processing.

2. the pre processing is much more resource intensive than visualization, and a visualization session can last hours as a user pokes around the UI; the cost savings on our AWS infrastructure is pretty big if we use a more modest instance

happy to hear thoughts/alternatives! we also wanted the visualizer to (eventually) be able to output things the user specified in the UI; we could unify the tools i suppose; something like:
- if user is preprocessing, output an archive in addition to other output
    - user waits a while for the IT to start
- if the user provides a preprocessed archive, skip preprocessing
    - the user does not wait a while for the IT to start

imo it comes down to opinion/convention as to which route to take. i am happy to do whatever everyone feels is best, this is merely my minimum viable product pass :D

Reply:
Makes total sense! We should not waste a lot of memory and CPUs for the Visualisation. Thanks for the details!